docs/well-known.md

```diff
@@ -0,0 +1,49 @@
+# Well-Known
+
+The mapping for a server name to a hostname for federation is done via
+`SRV` records. Synapse v1.0 mandates valid X.509 certificates for the
+federation endpoint where the certificate must match the server name.
+
+This presents difficulties for hosted server offerings: ExampleCorp
+may want to delegate responsibility for running its Matrix homeserver to an
+outside supplier, but it may be difficult for that supplier to obtain a TLS
+certificate for `exmaple.com` (and ExampleCorp may be reluctant to let them have
+one).
+
+`.well-known` solves this problem by augmenting the current `SRV` record
+with a `.well-known` lookup.
+
+## Configuring `.well-known`
+
+ 1. Give synapse a certificate corresponding to the target domain
+    (`customer.example.com`). Currently Synapse's ACME
```

Comment:
Hmm, this feels a bit harder to follow than it could be otherwise since ExampleCorp supposedly owns `example.com` but then that Matrix HS supplier controls `customer.example.com` if I am following, which seems unlikely... Perhaps `customer.example.com` should be `customer.homeservers.com` or really anything other than `*.example.com`, so it's clear that they are controlled by different parties?